### PR TITLE
[MU4] Network manager

### PR DESCRIPTION
**Pull request by RomanPudashkin**
- implemented head, post, put and delete requests

### CODE REVIEW

File: framework/network/internal/networkmanager.h [hunk 1]
```diff
@@ -22,35 +22,53 @@
...
 {
     Q_OBJECT
+
 public:
     explicit NetworkManager(QObject* parent = nullptr);
-    virtual ~NetworkManager() override;
+    ~NetworkManager() override;
 
     Ret get(const QUrl& url, QIODevice* incommingData) override;
+    Ret head(const QUrl &url) override;
```
Comment (Eism) [inline framework/network/internal/networkmanager.h:39]: Need to correct the style code

Comment (RomanPudashkin) [reply framework/network/internal/networkmanager.h]: fixed

File: framework/network/inetworkmanager.h [hunk 2]
```diff
@@ -35,8 +34,13 @@ class INetworkManager
     virtual ~INetworkManager() = default;
 
     virtual Ret get(const QUrl& url, QIODevice* incommingData) = 0;
+    virtual Ret head(const QUrl& url) = 0;
+    virtual Ret post(const QUrl& url, QIODevice* outgoingData, QIODevice* incommingData) = 0;
+    virtual Ret put(const QUrl& url, QIODevice* outgoingData, QIODevice* incommingData) = 0;
+    virtual Ret del(const QUrl& url, QIODevice* incommingData) = 0;
 
     virtual async::Channel<Progress> downloadProgressChannel() const = 0;
+    virtual async::Channel<Progress> uploadProgressChannel() const = 0;
```
Comment (igorkorsukov) [inline framework/network/inetworkmanager.h:43]: Maybe better one channel "progress"?

File: framework/network/internal/networkmanager.cpp [hunk 3]
```diff
@@ -58,21 +87,53 @@ Ret NetworkManager::get(const QUrl& url, QIODevice* incommingData)
...
+        prepareReplyTransmit(reply);
+    }
+
+    if (incommingData) {
+        prepareReplyReceive(reply, m_incommingData);
+    }
+
+    Ret ret = waitForReplyFinished(reply, TIMEOUT_MS);
+    if (!ret) {
+        LOGE() << "Error exec request, error" << ret.code() << ret.text();
```
Comment (igorkorsukov) [inline framework/network/internal/networkmanager.cpp:104]: Ret has a method `toString`

Comment (RomanPudashkin) [reply framework/network/internal/networkmanager.cpp]: fixed

File: framework/network/inetworkmanager.h (hunk 2 above)
Comment (RomanPudashkin) [reply framework/network/inetworkmanager.h]: fixed

**Review by igorkorsukov — APPROVED** [review]

**Review by Eism — APPROVED** [review]

**Pull request merged**